pkg/k8s/k8s_cluster.go

```diff
@@ -449,3 +489,33 @@ func (k *K8sCluster) ProxyGet(scheme, namespace, name, port, path string, params
 	}
 	return ret.Stream(k.ctx)
 }
+
+func (k *K8sCluster) GetObjectStatus(ref k8s.ObjectRef) (string, error) {
+	o, _, err := k.GetSingleObject(ref)
+	if o == nil {
+		return "", err
+	}
+	conditions, ok, err := o.GetNestedField("status", "conditions")
+	if !ok {
+		return "", err
+	}
+	last := conditions.([]interface{})[0].(map[string]interface{})["status"]
```

Comment:
This will crash in many situations, as the condition at index 0 is not always existing or the one you're looking for. You must iterate through all conditions and look for the one with type == "Ready" and then use that one.